src/NuGet.Core/NuGet.Protocol/Plugins/PluginCacheEntry.cs

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using NuGet.Common;
+
+namespace NuGet.Protocol.Plugins
+{
+    public sealed class PluginCacheEntry
+    {
+        public PluginCacheEntry(string rootCacheFolder, string pluginFilePath, string requestKey)
+        {
+            RootFolder = Path.Combine(rootCacheFolder, CachingUtility.RemoveInvalidFileNameChars(CachingUtility.ComputeHash(pluginFilePath)));
+            CacheFileName = Path.Combine(RootFolder, CachingUtility.RemoveInvalidFileNameChars(requestKey) + ".dat");
+            NewCacheFileName = CacheFileName + "-new";
+        }
+
+        internal TimeSpan MaxAge { get; set; } = TimeSpan.FromDays(30);
+        internal string CacheFileName { get; }
+        private string RootFolder { get; }
+        private string NewCacheFileName { get; }
+
+        public IReadOnlyList<OperationClaim> OperationClaims { get; private set; }
+
+        public void LoadFromFile()
+        {
+            Stream content = null;
+            try
+            {
+                content = CachingUtility.ReadCacheFile(MaxAge, CacheFileName);
+                if (content != null)
+                {
+                    ProcessContent(content);
+                }
+            }
+            finally
+            {
+                content?.Dispose();
+            }
+        }
+
+        private void ProcessContent(Stream content)
+        {
+            var serializer = new JsonSerializer();
+            using (var sr = new StreamReader(content))
+            using (var jsonTextReader = new JsonTextReader(sr))
+            {
+                OperationClaims = serializer.Deserialize<IReadOnlyList<OperationClaim>>(jsonTextReader);
+            }
+        }
+
+        public void AddOrUpdateOperationClaims(IReadOnlyList<OperationClaim> operationClaims)
```

Comment:
`OperationClaims` is defined as private set but also a public api to set it then why private set?

Reply:
good catch..had more logic there originally...it was autoupdating the file, but i decided to do it explicitly.